lib/std/os/linux.zig

```diff
@@ -1227,3 +1227,464 @@ test "" {
+        cgroup_getsockopt,
+        cgroup_setsockopt,
+        trace_raw_tp,
+        trace_fentry,
+        trace_fexit,
+        modify_return,
+        lsm_mac,
+        trace_iter,
+        cgroup_inet4_getpeername,
+        cgroup_inet6_getpeername,
+        cgroup_inet4_getsockname,
+        cgroup_inet6_getsockname,
+        xdp_devmap,
+    };
+
+    pub const Cmd = enum(usize) {
+        map_create,
+        map_lookup_elem,
+        map_update_elem,
+        map_delete_elem,
+        map_get_next_key,
+        prog_load,
+        obj_pin,
+        obj_get,
+        prog_attach,
+        prog_detach,
+        prog_test_run,
+        prog_get_next_id,
+        map_get_next_id,
+        prog_get_fd_by_id,
+        map_get_fd_by_id,
+        obj_get_info_by_fd,
+        prog_query,
+        raw_tracepoint_open,
+        btf_load,
+        btf_get_fd_by_id,
+        task_fd_query,
+        map_lookup_and_delete_elem,
+        map_freeze,
+        btf_get_next_id,
+        map_lookup_batch,
+        map_lookup_and_delete_batch,
+        map_update_batch,
+        map_delete_batch,
+        link_create,
+        link_update,
+        link_get_fd_by_id,
+        link_get_next_id,
+        enable_stats,
+        iter_create,
+    };
+
+    const obj_name_len = 16;
+    /// struct used by Cmd.map_create command
+    pub const MapCreateAttr = extern struct {
+        /// one of MapType
+        map_type: u32,
+        /// size of key in bytes
+        key_size: u32,
+        /// size of value in bytes
+        value_size: u32,
+        /// max number of entries in a map
+        max_entries: u32,
+        /// .map_create related flags
+        map_flags: u32,
+        /// fd pointing to the inner map
+        inner_map_fd: u32,
+        /// numa node (effective only if MapCreateFlags.numa_node is set)
+        numa_node: u32,
+        map_name: [obj_name_len]u8,
+        /// ifindex of netdev to create on
+        map_ifindex: u32,
+        /// fd pointing to a BTF type data
+        btf_fd: u32,
+        /// BTF type_id of the key
+        btf_key_type_id: u32,
+        /// BTF type_id of the value
+        bpf_value_type_id: u32,
+        /// BTF type_id of a kernel struct stored as the map value
+        btf_vmlinux_value_type_id: u32,
+    };
+
+    /// struct used by Cmd.map_*_elem commands
+    pub const MapElemAttr = extern struct {
+        map_fd: u32,
+        key: u64,
+        result: extern union {
+            value: u64,
+            next_key: u64,
+        },
+        flags: u64,
+    };
+
+    /// struct used by Cmd.map_*_batch commands
+    pub const MapBatchAttr = extern struct {
+        /// start batch, NULL to start from beginning
+        in_batch: u64,
+        /// output: next start batch
+        out_batch: u64,
+        keys: u64,
+        values: u64,
+        /// input/output:
+        /// input: # of key/value elements
+        /// output: # of filled elements
+        count: u32,
+        map_fd: u32,
```

Comment:
should this be `fd_t`?

Reply:
Similar to the `pid_t` bellow, the bpf header specified this as `u32`, do we prefer to use zig os types as often as we can?